### PR TITLE
string.h: strlcat() and strlcpy() tests

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: PD-189

## Description

Testing `strlcat` and `strlcpy` implementations

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (local tests testing on ia320generic qemu).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate. This PR is scheduled to merge **after**:
   - https://github.com/phoenix-rtos/libphoenix/pull/164

### CODE REVIEW

File: libc/string.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+TEST_TEAR_DOWN(string_strlcpy)
+{
+}
+
+
+TEST(string_strlcpy, strlcpy_fullcopy)
+{
+	/* Test full copy */
+	i = strlcpy(dest, source, 5);
```
Comment (agkaminski) [inline libc/string.c:53]:
```suggestion
	i = strlcpy(dest, source, sizeof(dest));
```

File: libc/string.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	TEST_ASSERT_EQUAL_INT(4, i);
```
Comment (agkaminski) [inline libc/string.c:54]:
```suggestion
	TEST_ASSERT_EQUAL_INT(sizeof(source) - 1, i);
```

File: libc/string.c [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	TEST_ASSERT_EQUAL_STRING(source, dest);
+}
+
+
+TEST(string_strlcpy, strlcpy_shorter)
+{
+	/* Test shorter than source copy */
+	i = strlcpy(dest, source, 3);
+	TEST_ASSERT_EQUAL_STRING("ab", dest);
+	TEST_ASSERT_EQUAL_INT(4, i);
```
Comment (agkaminski) [inline libc/string.c:64]:
```suggestion
	TEST_ASSERT_EQUAL_INT(sizeof(source) - 1, i);
```

File: libc/string.c [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+
+TEST(string_strlcpy, strlcpy_longer)
+{
+	/* Test longer than source copy */
+	source[3] = '\0'; /* source is now "abc" null terminated; */
+	i = strlcpy(dest, source, 5);
```
Comment (agkaminski) [inline libc/string.c:72]:
```suggestion
	i = strlcpy(dest, source, sizeof(dest));
```

File: libc/string.c (hunk 3 above)
Comment (agkaminski) [reply libc/string.c:64]: etc for other cases

File: libc/string.c [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+char dest[5];
+char tmplt[5];
+
+/* fixtures for strlcat */
+char source1[4];
+char source2[6];
+char buffer[12];
+
+/* common fixtures */
+int i;
```
Comment (agkaminski) [inline libc/string.c:31]: Common iterator name for global variable - I'm not sure if I like it (I think I don't). Maybe `retval`?

File: libc/string.c [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+TEST_GROUP(string_strlcat);
+
+
+TEST_SETUP(string_strlcat)
+{
+	memcpy(source1, "abc", 4);
+	memcpy(source2, "defgh", 6);
+	memcpy(buffer, "klmnopqrstu", 12);
+	i = -1;
```
Comment (agkaminski) [inline libc/string.c:115]: I don't think we need this - it's always assigned by each test

File: libc/string.c (hunk 5 above)
Comment (agkaminski) [reply libc/string.c:31]: Or maybe it should be a local variable (along with the name change)

**Review by agkaminski — COMMENTED** [review]
My comments about `sizeof` use are somewhat optional - it would make test more easily changeable in the future and also point out from where this value comes from - right now these are somewhat magic-numbery.

File: libc/string.c (hunk 5 above)
Comment (niewim19) [reply libc/string.c]: Variable left as global. Name changed to `retval` as suggested. Thanks!

File: libc/string.c (hunk 6 above)
Comment (niewim19) [reply libc/string.c]: Yes, you are right. Removed `i = -1`. Thanks!

**Comment by niewim19** [comment]
@agkaminski Thanks for review, all suggestions added, used `sizeof()` with correct variable where necessary.

File: libc/string.c [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+int retval;
+
+
+TEST_GROUP(string_strlcpy);
+
+
+TEST_SETUP(string_strlcpy)
+{
+	memcpy(source, "abcd", sizeof(source));
+	memcpy(dest, "xxxx", sizeof(dest));
```
[the remark this answers is not in the document]
Comment (niewim19) [reply libc/string.c]: Used `#defines` with sizes, thanks!

**Pull request merged**